assets/base.css

```diff
@@ -2225,15 +2225,15 @@ input[type='checkbox'] {
 }
 
 .shopify-section-header-hidden {
-  transform: translateY(-100%);
+  top: calc(-1 * var(--header-height));
```

Comment:
Ideally for animation we want to keep using `transform` as it's much more performant and doesn't trigger repaints from the browser. 
This is an interesting article about animation: https://web.dev/animations-guide/

But I wonder what other approach we could have to fix the issue 🤔

Reply:
I'm open to any suggestions but unfortunately I couldn't find any other workarounds. Also, as far I have investigated this topic, using both `transform` and `background-attachment` will always lead to a conflict we have.

Reply:
Yeah I'm not seeing any other way to fix it 🤔 
I kind of wonder if we should fix it 😅 Since we have different header styles now. People can work around it by setting the sticky header to always on or not sticky. 

Ideally we keep the best approach in place. As this change applies to all merchants when it's probably a smaller amount using gradients on their theme. 

For once, it's not an issue on Safari (desktop) but it is on Chrome and Firefox.

Reply:
I just tested it in Safari and the issue is there. In my opinion, the bug is pretty visible. I decided to spend a bit more time on finding another workaround. 

As an argument on your opinion I can say that, on one hand, you are absolutely right about the best practice but, on the other hand, this animation itself is quite simple and small to worry about what approach we take.

Reply:
Are you sure that you are using main and not this PR for testing? Because for me it's still happening in Safari ([video](https://screenshot.click/13-50-i65xc-725nw.mp4)).

Reply:
Ludo's article shows some tips on how to measure the impact of an animation on rendering using [Chrome devtools](https://web.dev/animations-guide/#layout-chrome). I think that we can start by comparing the use of transform vs top in Dawn's header to understand what the cost is (how much more time in ms is the browser spending on re-rendering)? If we observe >1000ms rendering for the animation to complete, then I'd say we need to look into more performant options.

Reply:
I did tests and it looks like the difference is fairly small, it's just around 16ms.  

<details><summary>Using `transform`</summary>
<img width="1201" alt="Screenshot 2023-01-16 at 4 09 56 PM" src="https://user-images.githubusercontent.com/105315663/212787373-d38f7079-0ef4-4917-b536-765d662140ba.png">
</details>

<details><summary>Using `top`</summary>
<img width="1241" alt="Screenshot 2023-01-16 at 4 11 22 PM" src="https://user-images.githubusercontent.com/105315663/212787403-6dcfd5a0-7c63-4775-8f39-cf3790d29855.png">
</details>

Reply:
ill try and take some time to test those changes on my old Pixel 2 to see if anything is noticeable.

Reply:
@eugenekasimov I tried to replicate your results and indeed it seems both animations complete in under ~150ms which is what we set the animation duration to. So maybe looking at total animation duration like i suggested isn't the most useful 🤔 I looked at some underlying numbers (Scripting, Rendering, Painting, System) and it does appear that the "top" animation requires 2-3x the time for those. I think that does indicate a drop in performance, but I don't think it's bad enough to block this PR.

![Screenshot 2023-01-17 at 4 13 14 PM2](https://user-images.githubusercontent.com/3710548/213015926-52375192-445b-4297-9877-80c5f62dc4eb.png)

Reply:
@KaichenWang thank you for the opinion. 

Btw, it seems you did not apply gradient for the header when you tested `transform` which might affect numbers :)

Reply:
Good point! Just re-ran the test. In general, the numbers can fluctuate a bit, but overall I'm observing the same trend.

![Screenshot 2023-01-18 at 10 14 28 AM 2](https://user-images.githubusercontent.com/3710548/213209607-fa4b18c1-5709-4bb0-a86c-256854e08941.png)